### ISSUE TITLE
python.install_env cannot be set to "venv" unless you are in a venv!

### ISSUE THREAD

**Issue by intractabilis** [mesonbuild/meson#12141]
**Describe the bug**
I cannot configure a project when it has `python.install_env=venv` 

**To Reproduce**
I cannot reliably reproduce it. However, I think I established that it never works with conda. Try this. Install Miniconda:
```
$ curl https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -o Miniconda3-latest-Linux-x86_64.sh
$ bash Miniconda3-latest-Linux-x86_64.sh
```
Reload your terminal, and create and activate a virtual environment
```
$ conda create --name meson-bug python=3.9
$ conda activate meson-bug
```
Create this `meson.build`
```
project('meson-bug')

py_mod = import('python')
py     = py_mod.find_installation()
```
Try to configure
```
$ meson setup -Dpython.install_env=venv build/debug
The Meson build system
Version: 1.2.1
Source dir: somepath/meson-bug
Build dir: somepath/meson-bug/build/debug
Build type: native build
Project name: meson-bug
Project version: undefined
Host machine cpu family: x86_64
Host machine cpu: x86_64

meson.build:4:16: ERROR: python.install_env cannot be set to "venv" unless you are in a venv!
```
I tried installing Meson to the virtual environment with
```
pip install meson
```
The result is the same. If I remove
```
if not self.info['is_venv']:
    raise mesonlib.MesonException('python.install_env cannot be set to "venv" unless you are in a venv!')
```
then everything works as expected, I didn't observe any problems.

**Expected behavior**
I expect the setup to finish without errors.

**system parameters**
* It's a plain native build.
* Manjaro Linux 23.0.0
* Python 3.9.
* Meson 1.2.1
* Ninja 1.11.1

**Comment by QuLogic** [comment]
> `$ conda create --name meson-bug python=3.9 $ conda activate meson-bug`

A `conda` environment is not the same as a `venv`-based virtual environment. So I'm not sure this is a bug, but a feature request to work with conda.

**Comment by eli-schwartz** [comment]
```python
'is_venv': sys.prefix != variables['base_prefix']
```

Apparently conda environments aren't the same... :)

Does the "system" option install to the created environment or to the conda base install?

**Comment by intractabilis** [comment]
> > `$ conda create --name meson-bug python=3.9 $ conda activate meson-bug`
> 
> A `conda` environment is not the same as a `venv`-based virtual environment. So I'm not sure this is a bug, but a feature request to work with conda.

Conda is a normal virtual environment. I gave a conda example just because it is reproducible 100%. My colleague and I observed the same behavior on MinGW platform with an ordinary, mundane virtual environment created with `venv`. It's just sometimes it works on MinGW. So, the bug has nothing to do with conda and is not limited to conda.

**Comment by intractabilis** [comment]
> Does the "system" option install to the created environment or to the conda base install?

I don't understand the question.

**Comment by dnicolodi** [comment]
You may want to have a look at `meson-python`. It works just fine in conda environments.

**Comment by intractabilis** [comment]
Again, I gave conda as just a way to reproduce the problem. Friendly speaking, I don't care about conda, and I don't use it. I observe this bug in other situations that are critical to me, and they are not conda.

**Comment by intractabilis** [comment]
> @intractabilis it would help if you could answer this question

What is "conda base install"?

**Comment by its-darsh** [comment]
I'm having the exact same issue with Python's venv, it keeps spitting that error even though I'm inside a venv.

Any fix?

**Comment by kaimast** [comment]
I just ran into this again, and it took me a while to remember why.

Make sure you run meson setup while in the venv and that meson is installed in the venv as well. If not, the ninja file might point to the wrong meson script.

**Comment by its-darsh** [comment]
I was able to fix my issue by running meson via `pip install`. that's the only way seems to be actually working

**Comment by kaimast** [comment]
> I was able to fix my issue by running meson via `pip install`. that's the only way seems to be actually working

Yes. That's what I meant with "meson is installed in the venv". Glad you figured it out!

**Comment by michaelgruner** [comment]
I had/have the same. I confirm using the meson installed in that environment works.

**Comment by tjwilli58** [comment]
I'm trying to build gstreamer using meson and I thought I'd try to set python.install_env from the default. I set it to `auto`, but some of the subprojects still didn't use the python in my venv, so I tried to set python.install_env=venv and got the error. 

I am running meson in a virtualenv, and also did a `pip install meson` in that environment. I do have conda installed, but in my PowerShell, I do a `conda deactivate` before I activate my venv. I have Python 3.12 and 3.11 installed in my %LOCALAPPDATA%.

```powershell
PS C:\Users\Tim> py -0
 -V:3.12 *        Python 3.12 (64-bit)
 -V:3.11          Python 3.11 (64-bit)
```

**Comment by tjwilli58** [comment]
Never mind about that. Setting to `auto` caused my `meson install` to put my `site-packages` in my `system` location instead of `prefix`.  Back to the default `prefix` setting for me.

**Comment by tjwilli58** [comment]
To have all of the subprojects find the same python, I needed to set up my `py.ini` file in $env:LOCALAPPDATA.
```
[defaults]
python=3
python3=3.12
```
